src/cgame/cg_predict.cpp

```diff
@@ -51,35 +49,23 @@ It also adds content flags to allow for more specific traces in synchronized cod
 */
 void CG_BuildSolidList()
 {
-	centity_t     *cent;
-	snapshot_t    *snap;
-	entityState_t *ent;
+	cg_solidEntities.clear();
```

Comment:
Should you reserve(MAX_ENTITIES_IN_SNAPSHOT]

Reply:
I don't know. On one side this means that we need to use `(sizeof(*centity_t) = 2448) * 1024 * 2` entities, that's 5,0MB(!) of `.bss` (heap) memory which is typically not used (only a few entities, while most of those entities are not used because they are not buildables). On the flip side this avoids possible out-of-memory errors.

Reply:
Oh wait, this is a `centity_t*`, so it has a size of 8. So that means 1,6kB. That's actually reasonable.